### DISCUSSION TITLE
Lista de PSPs recebedores PIX API

### DISCUSSION THREAD

**Discussion by rodrigonunes100** [bacen/pix-api#187] [Show and tell]
Existe alguma lista de PSPs recebedores que estão disponibilizando a PIX API  ? Seria interessante essa lista. Em muitos sites de PSPs recebedores não existe essa informação,  O ônus ficou com o empreendedor e desenvolvedor buscar essa informação. Vamos começar:

GerenciaNet - SIM
Cielo - ? - Dizem que estão implementando, mas não divulgam a data.
Santander - ?
Bradesco - ?
Itaú - ?
Iti Itáu - ?
BB - ?
PagSeguro - ?
BS2 -  SIM
Nubank - ?
MercadoPago - ?
C6 - ?
Caixa - ?
PicPay - ?
Algum outro ?

**Comment by rubenskuhl** [comment]
Mantida pelo BACEN, não, que já informou não imaginar fazer uma lista assim. Eu tenho mantido o #76 atualizado.

**Reply by GuerreiroPUSHINPAY** [reply]
Aproveito para apresentar uma ótima solução para Gateway de pagamento com a PUSHIN PAY
https://doc.pushinpay.com.br/#autenticacao
Melhor taxa de mercado.
Entre em contato conosco 
lorival.guerreiro@pushinpay.com.br

**Comment by rodrigonunes100** [comment]
Obrigado @rubenskuhl . O MercadoPago e PagSeguro estão disponibilizando a PIX API? não vejo nenhuma informação em nenhum site dos citados.  Em https://dev.pagseguro.uol.com.br/ do PagSeguro não encontramos nenhuma referencia ao PIX , em https://www.mercadopago.com.br/developers/pt/guides do mercado pago também não. No Bradesco não acho nenhuma informação

**Comment by rubenskuhl** [comment]
> Obrigado @rubenskuhl . O MercadoPago e PagSeguro estão disponibilizando a PIX API? não vejo nenhuma informação em nenhum site dos citados. Em https://dev.pagseguro.uol.com.br/ do PagSeguro não encontramos nenhuma referencia ao PIX , em https://www.mercadopago.com.br/developers/pt/guides do mercado pago também não. No Bradesco não acho nenhuma informação

PagSeguro disponibilizou um link no getpostman com a documentação, e uma conexão com uma API mock-up; já pedi uma atualização para eles e ainda não tenho um posicionamento atualizado. 

MercadoPago disse que teria API, mas como eles cobram por % e não quiseram fazer um acordo específico, não demos continuidade. Então até nova informação os manterei listados. 

[Manual_API_Recebimentos_PIX_V2.pdf](https://github.com/bacen/pix-api/files/5553711/Manual_API_Recebimentos_PIX_V2.pdf)
Bradesco eu recebi via gerente de relacionamento, mas imagino que se você pedir para suporte.api@bradesco.com.br eles te enviem. O documento se chama "Manual e Layout API de Recebimentos Pix". Anexei aqui o que eu tinha, mas fica o disclaimer de que eles podem e devem ter uma versão mais atualizada.

**Reply by SamuelCamposDF** [reply]
https://exclusive.bradesco/pix/assets/docs/api_pix_200.pdf

**Reply by rubenskuhl** [reply]
> https://exclusive.bradesco/pix/assets/docs/api_pix_200.pdf

Mas é a mais atual ? Já tem mais de um ano o arquivo...

**Comment by rodrigonunes100** [comment]
@rubenskuhl . Valeu.

**Comment by masyaf** [comment]
Itaú já possui API na v2, estamos utilizando inclusive em nosso ambiente de testes.

**Comment by rubenskuhl** [comment]
> Itaú já possui API na v2, estamos utilizando inclusive em nosso ambiente de testes.

O que me reportaram foram algumas divergências da API deles para a V2.1.2 como sistemas de credenciais, métodos e headers. Por enquanto estou imaginando apenas os escopos cob, pix e webhook como mandatórios. Ou seja, suporte mínimo a cobranças imediatas, tendo ou não suporte a cobranças com vencimento e em lote, mas já com configuração de webhook por chave.

**Comment by flaviolenz** [comment]
Nao vejo problema cobrar em % do valor da transacao.
Os cartoes hoje fazem assim e se for pra substitui-los, pode ficar extremamente interessante.
Cobrar valor fixo eh que inviabiliza usar PIX pra recebimento em negocios onde o ticket medio eh pequeno.

**Comment by flaviolenz** [comment]
> [Manual_API_Recebimentos_PIX_V2.pdf](https://github.com/bacen/pix-api/files/5553711/Manual_API_Recebimentos_PIX_V2.pdf)

Alguem chegou a evoluir algo aqui?
Olha o mTLS, que pesado!!!

No caso, certificado auto-assinado eh dureza.
Porque nao fez logo pela ICP-Brasil.
Toda empresa ja tem um e faz conexao com a SEFAZ pra emitir NF-e.

Nesse caso ai tem que ser um por estabelecimento ou pode ser um pro ERP e com ele, se conectar pra todos os estabelecimentos?

**Comment by rubenskuhl** [comment]
> Nao vejo problema cobrar em % do valor da transacao.
> Os cartoes hoje fazem assim e se for pra substitui-los, pode ficar extremamente interessante.
> Cobrar valor fixo eh que inviabiliza usar PIX pra recebimento em negocios onde o ticket medio eh pequeno.

Isso foi endereçado por alguns operadores colocando um preço fixo em função do valor, limitando a um valor máximo. Um exemplo: https://gerencianet.com.br/tarifas/ .

Notar que % baixo talvez até fosse factível, mas 1% dói.

**Comment by rubenskuhl** [comment]
> > [Manual_API_Recebimentos_PIX_V2.pdf](https://github.com/bacen/pix-api/files/5553711/Manual_API_Recebimentos_PIX_V2.pdf)
> 
> Alguem chegou a evoluir algo aqui?

Teve gente sim que conseguiu integrar. Não foi o meu caso então deixo para quem foi se manifestar, caso queira. 

> Olha o mTLS, que pesado!!!

O mTLS padronizado é pesado, mas o descrito aí é pior mesmo. 

> No caso, certificado auto-assinado eh dureza.

Isso é um dos motivos pelo qual listei o Bradesco em #76 como tendo API proprietária, pois o manual do Pix claramente veda isso. 

> Porque nao fez logo pela ICP-Brasil.
> Toda empresa ja tem um e faz conexao com a SEFAZ pra emitir NF-e.

Isso não funciona tão bem assim... o certificado A-3 é um pesadelo de interoperabilidade para sistemas automatizados. Usávamos uma interface de boletos registrados que exigia ICP-Brasil, por sinal a do próprio Bradesco, e foi um alívio migrar para outra. 

> Nesse caso ai tem que ser um por estabelecimento ou pode ser um pro ERP e com ele, se conectar pra todos os estabelecimentos?

Um por correntista. Há um issue propondo Authorization Credential Flow que talvez enderece melhor um terceiro autorizado a saber informação de recebimento.

**Comment by flaviolenz** [comment]
Muita gente paga mais que isso de Debito.
E de credito, eh comum pagar mais de 2%.
E ainda recebe so no dia seguinte.

Com a concorrencia, tende a baixar MUITO.

**Comment by rubenskuhl** [comment]
> Muita gente paga mais que isso de Debito.
> E de credito, eh comum pagar mais de 2%.
> E ainda recebe so no dia seguinte.
> 
> Com a concorrencia, tende a baixar MUITO.

As bandeiras de cartão tem muitas bocas para alimentar, o objetivo do Pix é exatamente diminuir isso e permitir uma operação mais eficiente de comércio e serviços. Quem quiser cobrar percentagem, cobrará assim e o mercado irá dizer como prefere. Tanto que mantenho prestadores com cobrança de % em #76 , esse caso específico foi da minha experiência e em as pessoas sabendo que opções tem e quanto cada uma custa, tomarão suas decisões.

**Comment by pnegri** [comment]
Só um ponto pessoal, dependendo do ticket médio, a % passa a ser diversas vezes menor que uma cobrança de centavos. No geral a % passa a não ser interessante com tickets médios altos. Com um ticket de 50 reais, não é um absurdo deixar 1% na mesa. Existe um custo aí de infraestrutura, links, custo por banda na própria mensageria do bacen que não torna a transação tão de graça assim. Muita gente olha só pro 1 cent de custo de 10 transações, mas esquece de todo o custo regulatório para manter uma IP ou IF, o custo de toda parte de rede (Que é pesado), e afins. Só pra ter uma idéia, iugu demorou 3 anos para conseguir a licença, e o custo para manter a infra sem usar ultrapassava 100 mil reais por mês, e tivemos que manter durante todo este tempo, isso sem falar do custo regulatório, compliance, auditoria interna, auditoria externa, controles internos etc.  E nem estou pondo na conta o custo de capital regulatório de IP (2% de tudo que entra e que sai). Só funciona na escala.

**Comment by rubenskuhl** [comment]
Quem quiser pagar 1% pode pagar se preferir, mas sugiro observar a lista do #76 e através de pesquisa de mercado diminuir bem em relação a isso. Tipo, 5 ou 10 vezes menos. A API padronizada é um forte indutor de competição, que os estabelecimentos comerciais podem e devem ter a seu favor.

**Comment by aifabio** [comment]
Palhaçada o BBC que bagunça está virando um monopólio ficando a mesma coisa, em vez de nos ajudar a integração estamos tendo que ficar dependente de toda forma e ainda dando mais trabalho

**Comment by ivent** [comment]
Estou aqui maluco já como O BBC lança uma coisa pra reduzir o custo obriga a todos os bancos a usar a api deles que já inibe que eles possam dificultar as coisas mas coloca essa parada de PSP webhooks como dependência? um serviço que poucas empresas no brasil oferecem, cara eu passei dias integrando a api no Pagseguro e no Juno consigo emitir os qrcode de pix etc.. mas não sei quem pagou pois precisa do PSP quanto custa o banco mesmo enviar um retorno que recebeu o pix sem isso não adianta nada vc automatizar o serviço se não pode saber quem pagou lamentável, as empresas parece não se importarem em facilitar isso pois eles querem cobrar para usarem o checkout deles onde eles cobram % pra pagar o pix que deveria ser grátis lamentável, atualmente consegui fazer a integração mas tenho que confirmar o pagamento manualmente entrando na conta, velho quanto custaria a api do BBC mandar um simples retorno de "PixPago" lamentável alguém tem alguma solução me ajude pvf :-(

**Reply by rubenskuhl** [reply]
A API da Juno tem apenas geração de QR-Code estático, não notificação. PagSeguro eu não olhei, tem o link para a doc ? 

Os PSPs listados em #76 tem API completa com as notificações via webhook, tal como Gerencianet, BS2, Itaú e Bradesco.

**Reply by ivent** [reply]
@rubenskuhl Obrigado eu estou falando com o pessoal da Juno para ver a viabilidade de ter este retorno, o pagseguro eu vi lá no postman que eles estão desenvolvendo mas o time de dev deles é lento e como não tem interesse pode ser que não avancem, eu estou quase fechando com a Gerencianet mesmo mas ainda irei aguardar uma solução para não depender de serviço de terceiros para apenas confirmar um pagamento de pix.

**Reply by rubenskuhl** [reply]
> @rubenskuhl Obrigado eu estou falando com o pessoal da Juno para ver a viabilidade de ter este retorno, o pagseguro eu vi lá no postman que eles estão desenvolvendo mas o time de dev deles é lento e como não tem interesse pode ser que não avancem, eu estou quase fechando com a Gerencianet mesmo mas ainda irei aguardar uma solução para não depender de serviço de terceiros para apenas confirmar um pagamento de pix.

Não tem como não ter serviço de alguém para notificar, os pagamentos acontecem dentro do sistema financeiro e alguém que participa precisa te avisar.

**Reply by ivent** [reply]
Sim claro mas o próprio banco mesmo não poderia avisar? pq precisa de um serviço de terceiro?

**Reply by rubenskuhl** [reply]
Mas é a instituição em que você tem conta que te avisa, não um terceiro. A API só te permite gerar cobranças e receber notificações de chaves Pix da conta em seu nome.

**Reply by ivent** [reply]
Rubens Obrigado pelos esclarecimentos, na real o que eu queria mesmo era ter um retorno de uma transferência pix via api já que toda propaganda  pix ninguém fala de pagar taxas porém eu entendi agora que nenhum banco ou fintech irá oferecer isso já que o BACEN deu este alivio pra eles aceitarem o Pix só pra transferências que são gratuitas e logo dar um retorno sobre elas iria possibilitar fazer pagamentos e Vendas via pix de forma gratuita também e como os bancos cobram taxas pra oferecer este serviço, logo não faz sentido dar o retorno pro modelo do pix gratis pra transferências me corrige se estiver errado meu raciocínio.  

Neste caso optei pela Juno pelo time de Dev deles que ajuda muito e pela api que é bem legal e consegui fazer uma integração com uma taxa fixa de R$ 2,99 por pagamento via pix dinâmico que se não me engano é o mesmo valor cobrado pela gerencianet já que tenho produtos que podem chegar até R$ 450 e o pagseguro cobra por 1,89% logo sairia mais caro pagar por porcentagem sem falar que não acho justo o intermediador receber porcentagem do seu trabalho duro e sim é justo pagar um valor fixo pelo serviço que eles prestam mas é isso fica a dica, quem tiver outra opção melhor e já tiver integrado algum sistema de pagamento via pix com uma taxa fixa menor pode colaborar também pra enriquecer este post.

**Reply by renatofrota** [reply]
@ivent ,

> Neste caso optei pela Juno pelo time de Dev deles que ajuda muito e pela api que é bem legal e consegui fazer uma integração com uma taxa fixa de R$ 2,99 por pagamento via pix dinâmico que se não me engano é o mesmo valor cobrado pela gerencianet já que tenho produtos que podem chegar até R$ 450

A Gerencianet cobra 0,99% por recebimento de QR dinâmico, com um **teto de R$ 0,89** (ou seja, **30%** do que você está pagando pra Juno).

**Reply by rubenskuhl** [reply]
> Rubens Obrigado pelos esclarecimentos, na real o que eu queria mesmo era ter um retorno de uma transferência pix via api já que toda propaganda pix ninguém fala de pagar taxas porém eu entendi agora que nenhum banco ou fintech irá oferecer isso já que o BACEN deu este alivio pra eles aceitarem o Pix só pra transferências que são gratuitas e logo dar um retorno sobre elas iria possibilitar fazer pagamentos e Vendas via pix de forma gratuita também e como os bancos cobram taxas pra oferecer este serviço, logo não faz sentido dar o retorno pro modelo do pix gratis pra transferências me corrige se estiver errado meu raciocínio.

O Banco Central sempre falou de isenção de taxas para pessoas físicas em situações não comerciais, informando que poderiam haver taxas tanto para pessoa jurídica, em qualquer caso, quanto para pessoa física, em situações comerciais. 
Então cobrar pela API que realiza cobranças está alinhado com tudo que sempre foi divulgado desde 12 meses atrás. 


> 
> Neste caso optei pela Juno pelo time de Dev deles que ajuda muito e pela api que é bem legal e consegui fazer uma integração com uma taxa fixa de R$ 2,99 por pagamento via pix dinâmico que se não me engano é o mesmo valor cobrado pela gerencianet já que tenho produtos que podem chegar até R$ 450 e o pagseguro cobra por 1,89% logo sairia mais caro pagar por porcentagem sem falar que não acho justo o intermediador receber porcentagem do seu trabalho duro e sim é justo pagar um valor fixo pelo serviço que eles prestam mas é isso fica a dica, quem tiver outra opção melhor e já tiver integrado algum sistema de pagamento via pix com uma taxa fixa menor pode colaborar também pra enriquecer este post.

O problema da Juno é que eles insistiram por muito tempo em uma API de cobrança proprietária para o Pix, apesar de inúmeros alertas de que isso ia contra o regulamento. Quando caíram em si, estavam meses atrasados no desenvolvimento de uma API padrão, coisa que Gerencianet tem, e que as APIs do BS2 e dos grandes bancos de varejo são ao menos próximas de ter. 

E como já citado pelo @renatofrota , o custo da Gerencianet para essa faixa de recebimento em baixo volume é de 90 centavos (R$0,01 da geração do QR-Code dinâmico, R$0,89 do recebimento). E se você tiver volume, consegue reduzir isso ainda mais. 

Apesar de não ser de graça, a cobrança via Pix é tipicamente mais barata que seus pares boleto e cartão.

**Reply by ivent** [reply]
Oi @renatofrota e @rubenskuhl ótimas explicações agora ficou muito claro pra mim, sobre o teto da Gerancianet realmente não sabia rpz poxa então vou considerar migrar para eles então, muito obrigado pelos esclarecimentos.

**Comment by claudiopolonia** [comment]
Pessoal o bradesco tem API PIX com geração e retono de pagamento? alguem tem documentação? Obrigado

**Reply by rubenskuhl** [reply]
Sim, vide #76 . Lá tem o e-mail do Bradesco para pedir a documentação, que não tem um site.

**Comment by duduccosta** [comment]
Alguém tem a documentação completa do Itaú?

**Reply by amarborto** [reply]
Quando fiz a integração eles liberaram o cadastro para efetuar o login em https://developer.itau.com.br/login 
para obter a doc. Entre em contato com seu gerente, provavelmente ele poderá liberar pra vc.

**Comment by Unbreak4ble** [comment]
Gostaria de saber se nesta existe suporte para o C6 Bank, se não existir suporte, conhecem alguma api que suporte?

**Reply by rubenskuhl** [reply]
> Gostaria de saber se nesta existe suporte para o C6 Bank, se não existir suporte, conhecem alguma api que suporte?

Na lista do issue #76 o C6 é mencionado no trecho de APIs não padrão.

**Comment by b2ubank** [comment]
Este banco tem API pix a partir de R$ 0,19 o pix

https://apidoc.b2ubank.com/

![image](https://github.com/bacen/pix-api/assets/137004747/c9611472-1c80-40cd-a605-96c72b730a91)

**Reply by rubenskuhl** [reply]
Esta lista é apenas de PSPs oferecendo API Pix, e a empresa em questão não é um PSP, ela usa um PSP. 
Notar que isso está em fase de regulamentação pelo Banco Central (gestor de pagamentos), e de uma hora para outra essa atividade pode estar vedada.

**Comment by pnegri** [comment]
Iugu tem a partir de 5 cents a tempos já. Abs

**Reply by rubenskuhl** [reply]
No caso acima é 19 centavos para 600 mil transações/mês... 5 centavos seria para quantas transações por mês no PSP Iugu ?

**Reply by pnegri** [reply]
10 estático, 13 sem vencimento, 16 com venc.

**Reply by b2ubank** [reply]
B2U Bank consegue chegar em 5 centavos, tudo depende de volume. Além disto, lembrem-se de pensar no estorno do pix (emv), no B2ubank ficamos do lado do nosso cliente, em outros aí, fazem o estorno, bloqueiam a conta e mandam se virar
